### ISSUE TITLE
SC1036: False positive using "let" with parentheses

### ISSUE THREAD

**Issue by jschleus** [koalaman/shellcheck#2214]
#### For bugs
- Rule Id:  SC1036
- My shellcheck version: "online"
- [x] The rule's wiki page does not already cover this

#### Here's a snippet or screenshot that shows the problem:
```
#!/bin/bash
TEST=1; A=1; B=2
if [ "$TEST" -eq 1 ]; then
  let C=(A+B-1)/B*B
fi
```
#### Here's what shellcheck currently says:
```
In line 3:
if [ "$TEST" -eq 1 ]; then
^-- SC1046: Couldn't find 'fi' for this 'if'.
^-- SC1073: Couldn't parse this if expression. Fix to allow more checks.

In line 4:
  let C=(A+B-1)/B*B
        ^-- SC1036: '(' is invalid here. Did you forget to escape it?
        ^-- SC1047: Expected 'fi' matching previously mentioned 'if'.
        ^-- SC1072: Expected 'fi'. Fix any mentioned problems and try again.
```
#### Here's what I wanted or expected to see:
No errors (like for the equivalent bash command
```
C=$(((A+B-1)/B*B)))
```
Parentheses seem to be allowed in `arithmetic expression`. At least the above code works as expected and at the end of the section `ARITHMETIC EVALUATION` the `bash` manual page says: `Sub-expressions in parentheses are evaluated first and may override the precedence rules above`.

The other errors are probably a consequence of the error SC1036.
